api/tests/Unit/Controller/SelfRegisterControllerTest.php

```diff
@@ -180,4 +190,66 @@ public function throwErrorForDuplicate()
             'ClientSecret' => API_TOKEN_DEPUTY,
         ]);
     }
+
+    /**
+     * @test
+     */
+    public function throwErrorForValidCaseNumberButDetailsNotMatching()
+    {
+        $casRec = new CasRec([
+            'Case' => '97643164',
+            'Surname' => 'Douglas',
+            'Deputy No' => 'DEP00199',
+            'Dep Surname' => 'Murphy',
+            'Dep Postcode' => 'SW1',
+            'Typeofrep' => 'OPG102',
+            'Corref' => 'L2',
+            'NDR' => 1,
+        ]);
+
+        $this->fixtures()->persist($casRec);
+        $this->fixtures()->flush();
+
+        $token = $this->login('deputy@example.org', 'DigidepsPass1234', API_TOKEN_DEPUTY);
+
+        $responseArray = $this->assertJsonRequest('POST', '/selfregister', [
+            'mustFail' => true,
+            'AuthToken' => $token,
+            'data' => [
+                'firstname' => 'Zac',
+                'lastname' => 'Tolley',
+                'email' => 'wrong@email.com',
```

Comment:
Worth using example.org/com here as it's reserved, rather than "fake" emails like this.